src/main/java/com/aws/greengrass/clientdevices/auth/connectivity/CISShadowMonitor.java

```diff
@@ -245,6 +248,24 @@ private synchronized void processCISShadow(String version, Map<String, Object> d
                 // We won't retry in this case, but we will update the CIS shadow reported state
                 // to signal that we have fully processed this version.
                 try {
+                    LOGGER.atInfo().kv(VERSION, version)
+                            .log("No connectivity info found. Skipping cert re-generation");
+                    updateCISShadowReportedState(desiredState);
+                } finally {
+                    // Don't process the same version again
+                    lastVersion = version;
+                }
+                return;
+            }
+
+            // skip cert rotation if connectivity info hasn't changed
+            List<String> cachedHostAddresses = connectivityInformation.getCachedHostAddresses();
+            if (Objects.equals(prevCachedHostAddresses, cachedHostAddresses)) {
```

Comment:
convert to sets before comparing, otherwise duplicates or changed order will mess things up